Healthchecks Reference documentation (#173)
syntasso/kratix-docs: docs/main/03-reference/16-health-checks/02-health-definition.md

---
description: Documentation for the Health Definition Custom Resources
title: Health Definition
sidebar_label: Health Definition
id: healthdefinition
---

# Health Definition

> When a Promise defines a Healthcheck, Kratix will automatically create a Health Definition for that Healthcheck. A Health Definition is the outline of the task that will be performed on a Destination to verify the health of a Resource Request.

Reviewer: nitpick: health definition gets automatically created for resource request, not just installing the Promise. Maybe we could make this a bit clearer?

Author: Not a nitpick, a very good point

```yaml
apiVersion: platform.kratix.io/v1alpha1
kind: HealthDefinition
metadata:
  name: healthdefinition
  namespace: default
spec:
  name: default-promise-name-resource-name
  # A reference to the Resource Request the Health Check should be performed against
  resourceRef:
    name: request-name
    namespace: default

>   # A reference the Promise the Health Check should be performed against  promiseRef:

Reviewer: ```suggestion
  # Name of the Promise that the Resource Request belongs to.
```

>   promiseRef:
>     name: promise-name
>     namespace: default

Reviewer: ```suggestion
  promiseRef:
    name: promise-name
```

  # The time or interval the check should run against
  # This can follow Cron syntax or macros such as @hourly
  schedule: "* * * * *"
  # The definition of the Resource the check will be performed against
  input: |
    apiVersion: mypromise.org/v1
    kind: someservice
    metadata:
        name: someservice
        namespace: default
    spec:
        example: data
    status:
        url: test.com
  # The task to be performed on the destination
  workflow:

>     # A Kratix provided Pipeline that runs an ordered set of OCI compliant images

Reviewer: ```suggestion
    # A Pipeline that runs an ordered set of OCI compliant images to perform health checks
```

    apiVersion: platform.kratix.io/v1alpha1
    kind: Pipeline
    metadata:
      name: health
    spec:
      containers:

>         - image: busybox

Reviewer: nitpick: maybe a more descriptive container image than busybox
